openlabcmd/openlabcmd/zk.py

```diff
@@ -84,19 +84,29 @@ def lost(self):
             return True
         return self.client.state == KazooState.LOST
 
-    def connect(self, hosts=None, read_only=False):
+    def connect(self, hosts=None, timeout=None, read_only=False):
         if not hosts:
             if not self.config:
                 raise exceptions.ClientError('Either config object or hosts '
                                              'string should be provided.')
-            else:
-                try:
-                    hosts = self.config.get('ha', 'zookeeper_hosts')
-                except configparser.NoOptionError:
-                    raise exceptions.ClientError(
-                        "The config doesn't [ha]zookeeper_hosts option.")
+            try:
+                hosts = hosts or self.config.get('ha', 'zookeeper_hosts')
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+
+        if not timeout:
+            timeout = self.config.get('ha', 'zookeeper_connect_timeout',
+                                      fallback=5)
+        try:
+            timeout = int(timeout)
```

Comment:
```python
>> int(-1)
-1
```
Maybe could add some limitation on positive value.

Reply:
I saw this is fixed  in the following line.